.github/workflows/build-magma-rocm-linux.yml

```
@@ -0,0 +1,69 @@
+name: build-linux-magma-rocm
+
+on:
+  push:
+    branches:
+      main
+    paths:
+      - .ci/magma-rocm/*
+      - .ci/magma-rocm/package_files/*
+      - .github/workflows/build-magma-rocm-linux.yml
+  pull_request:
+    paths:
+      - .ci/magma-rocm/*
+      - .ci/magma-rocm/package_files/*
+      - .github/workflows/build-magma-rocm-linux.yml
+
+defaults:
+  run:
+    shell: bash -x -e -l {0}
+env:
+  BUILD_ENVIRONMENT: build-linux-magma-rocm
+  IN_CI: 1
+  IS_GHA: 1
+
+concurrency:
+  group: ${{ github.workflow }}-${{ github.event.pull_request.number || github.sha }}-${{ github.event_name == 'workflow_dispatch' }}
+  cancel-in-progress: true
+
+jobs:
+  build-linux-magma-rocm:
+    if: github.repository_owner == 'pytorch'
+    runs-on: linux.12xlarge
+    permissions:
+      id-token: write
+    strategy:
+      matrix:
+        rocm_version: ["63", "624"]
+    steps:
+      - name: Checkout PyTorch
+        uses: actions/checkout@v4
+      - name: Build Magma Rocm
+        working-directory: .ci/magma-rocm
+        run: |
+          # Produces artifacts under magma-rocm/output/linux-64/magma-rocm*.bz2
+          make magma-rocm${{ matrix.rocm_version }}
+      - name: Save as artifact
+        uses: actions/upload-artifact@v4
+        with:
+          path: .ci/magma-rocm/output/linux-64/magma-rocm*.bz2
+          name: artifact_${{ matrix.rocm_version }}
+      - name: Configure AWS credentials(PyTorch account)
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/main' }}
+        uses: aws-actions/configure-aws-credentials@v3
```

new actionlint does not like v3 
```suggestion
        uses: aws-actions/configure-aws-credentials@v4
```